openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -120,15 +122,16 @@ public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, Strin
         return openapi;
     }
 
-    public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi) {
+    public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi) throws BallerinaOpenApiException {
```

Comment:
Do we handle this exception where we are supposed to use this function?

Reply:
This is thrown from the generator functions: [`generateClientAndService`](https://github.com/ballerina-platform/openapi-tools/blob/3eda060f1933149b197714927fd5627c50a7399f/openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java#L102)

In CLI, we capture this error, add a log and exit : https://github.com/ballerina-platform/openapi-tools/blob/3eda060f1933149b197714927fd5627c50a7399f/openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiCmd.java#L500